lib/resin-lint.ts

```diff
@@ -201,123 +209,113 @@ const runLint = async function(
+		);
+	}
 
-				// TSLint config needs to be loaded with `loadConfigurationFromPath`
-				// Coffeelint needs to be loaded as a plain file
-				let config: {} = typescriptCheck
-					? tslint.Configuration.loadConfigurationFromPath(
-							resinLintConfiguration.configPath,
-					  )
-					: parseJSON(resinLintConfiguration.configPath);
+	let configOverridePath;
+	// optimist converts all --no-xyz args to a argv.xyz === false
+	const prettierCheck = options.argv.prettier !== false;
+	const testsCheck = options.argv.tests === true;
+	const typescriptCheck = options.argv.typescript;
+	const autoFix = options.argv.fix === true;
+	const resinLintConfiguration = typescriptCheck
+		? prettierCheck
+			? configurations.typescriptPrettier
+			: configurations.typescript
+		: configurations.coffeescript;
+
+	if (options.argv.p) {
+		console.log(fs.readFileSync(resinLintConfiguration.configPath).toString());
+		process.exit(0);
+	}
 
-				if (options.argv.f) {
-					configOverridePath = fs.realpathSync(options.argv.f);
-				}
+	// TSLint config needs to be loaded with `loadConfigurationFromPath`
+	// Coffeelint needs to be loaded as a plain file
+	let config: {} = typescriptCheck
+		? tslint.Configuration.loadConfigurationFromPath(
+				resinLintConfiguration.configPath,
+		  )
+		: parseJSON(resinLintConfiguration.configPath);
 
-				if (!options.argv.i && !configOverridePath) {
-					configOverridePath = findFile(resinLintConfiguration.configFileName);
-				}
+	if (options.argv.f) {
+		configOverridePath = fs.realpathSync(options.argv.f);
+	}
 
-				if (configOverridePath) {
-					// Extend/override default config
-					if (typescriptCheck) {
-						const configOverride = tslint.Configuration.loadConfigurationFromPath(
-							configOverridePath,
-						);
-						config = tslint.Configuration.extendConfigurationFile(
-							config as tslint.Configuration.IConfigurationFile,
-							configOverride,
-						);
-					} else {
-						const configOverride = parseJSON(configOverridePath);
-						const { merge } = await import('lodash');
-						config = merge(config, configOverride);
-					}
-				}
+	if (!options.argv.i && !configOverridePath) {
+		configOverridePath = findFile(resinLintConfiguration.configFileName);
+	}
 
-				const paths = options.argv._;
+	if (configOverridePath) {
+		// Extend/override default config
+		if (typescriptCheck) {
+			const configOverride = tslint.Configuration.loadConfigurationFromPath(
+				configOverridePath,
+			);
+			config = tslint.Configuration.extendConfigurationFile(
+				config as tslint.Configuration.IConfigurationFile,
+				configOverride,
+			);
+		} else {
+			const configOverride = parseJSON(configOverridePath);
+			const { merge } = await import('lodash');
+			config = merge(config, configOverride);
+		}
+	}
 
-				resinLintConfiguration.prettierCheck = prettierCheck;
-				resinLintConfiguration.testsCheck = testsCheck;
-				return runLint(resinLintConfiguration, paths, config, autoFix);
-			})
-			.return();
```

Comment:
Not sure it matters: I think the last `.return()` in the old code would mean that the function returns `undefined`, and in the new code it returns what `runLin()` returns. Probably doesn't matter.

Reply:
It shouldn't matter but it's also completely unnecessary to return anything, and it was returning `process` - fixed it